### PR TITLE
Add: Test that directories can be sgid and not owned by root

### PR DESCRIPTION
**Pull request by nickanderson**
Ref: https://dev.cfengine.com/issues/7989

### CODE REVIEW

**Comment by nickanderson** [comment]
@kacf and or @jimis please see this somewhat related change as well: https://github.com/cfengine/core/pull/2538

File: tests/acceptance/10_files/unsafe/can_set_sticky_bits_with_nonroot_owner_and_group.cf [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+{
+
+  vars:
+      # We expect to find an ok_ prefixed class for each of the modes defined in
+      # init.
+      "expected_classes" slist => maplist("ok_$(this)", @(init.modes));
+
+      # Get the octal mode for each of the files
+      "result_permoct_$(init.modes)"
+        string => execresult('/usr/bin/stat -c %a $(G.testdir)/test_sgid/$(init.modes)', "useshell"),
```
Comment (kacf) [inline tests/acceptance/10_files/unsafe/can_set_sticky_bits_with_nonroot_owner_and_group.cf]: AIX (and possibly others) doesn't have this tool. Can you add it to the list of tools near the top of `dcs.cf.sub`, and then use `"test_skip_unsupported" string => "windows|!has_stat";` to skip it there?

**Comment by kacf** [comment]
Looks good otherwise.

**Comment by nickanderson** [comment]
@kacf updated as requested

File: tests/acceptance/10_files/unsafe/can_set_sticky_bits_with_nonroot_owner_and_group.cf (hunk 1 above)
Comment (kacf) [inline tests/acceptance/10_files/unsafe/can_set_sticky_bits_with_nonroot_owner_and_group.cf]: Sorry, I also meant use it here, as `$(G.stat)`.

**Comment by nickanderson** [comment]
superseded by #2545

**Pull request closed without merging**